### PR TITLE
Cypress Axe Serious Level Issue Fixes

### PR DESCRIPTION
**Pull request by sranc046**
Here are my suggested changes for our "serious" level issues. The main issues raised during the tests were related to colour contrast and nested interactive elements (regarding tooltips). After testing them out, can confirm that these issues do not break the Google Chrome screen reader or Mac VoiceOver.

- As far as I can tell, the colour contrast is a bug and only raises an issue during the tests. Does not seem to appear on the main site. Which means I haven't been able to find how to fix it. I believe it has something to do with positioning of certain elements at the time of the scan? Anyways, I suggest disabling this rule as I have already addressed the critical level version (which lead to some colour tweaking of small fonts on light backgrounds). I think this rule is "easy enough" to follow and apply without the need for a check during tests.

- Since (I think) we are dropping tooltips soon I suggest just excluding these elements from the scan. Replacements for these tooltips might be tricky if we are trying to follow this rule about interactive nested elements but at least they don't break the two screen readers mentioned above. If this problem arises again with the implementation, we always have the option of excluding them too.

Apologies for just a single commit, lots of testing back and forth around whole codebase, enabling and disabling of elements/rules, and just lost track of progress. Follow up PR for moderate level issues soon, mostly just consists of heading level semantics that are fiddly to adjust.

### CODE REVIEW

**Review by Stephen-ONeil — CHANGES_REQUESTED** [review]
Suggestions in Infographic.js should be dealt with~, comments on lack of granularity in controls for Axe can be a follow up issue/branch.~

**Comment by Stephen-ONeil** [comment]
- [ ] Also fix CI build only color-contrast on `.lang-switcher` by changing how that big red dev link banner looks (I don't remember how that dev link banner works though, haha)

File: client/cypress/integration/InfoBase/route_tests.spec.js
Comment (Stephen-ONeil) [reply client/cypress/integration/InfoBase/route_tests.spec.js]: Fixed with snapshot allow list

File: client/src/infographic/Infographic.js
Comment (sranc046) [reply client/src/infographic/Infographic.js:226]: So the problem I was trying to resolve here was that there is an error when applying an aria-label to a div that has no content (ex: infographic/dept/326/all_data).

Comment (Stephen-ONeil) [reply client/src/infographic/Infographic.js:226]: Haha, that one's a bug https://github.com/TBS-EACPD/infobase/issues/1369

File: .eslintignore
Comment (Stephen-ONeil) [inline .eslintignore:9]: Minor rebase cleanup, this line was dropped on master I believe. Can delete it here too (if you have a local `transpiled_build` directory anywhere, you can delete it too)

File: client/src/components/CheckBox/CheckBox.tsx
Comment (Stephen-ONeil) [inline client/src/components/CheckBox/CheckBox.tsx:78]: Oh, this can be cleaned up too.

File: client/src/components/VisibilityControl/VisibilityControl.tsx
Comment (Stephen-ONeil) [inline client/src/components/VisibilityControl/VisibilityControl.tsx:42]: Another unused `data-cy`

**Review by Stephen-ONeil — APPROVED** [review]
Very minor cleanup points, please address those before merging, but pre-approved to go once you've done that :+1:

**Pull request merged**
